lib/srv/db/cloud/resource_checker_url_aws.go

```diff
@@ -0,0 +1,238 @@
+	}
+	return trace.Wrap(c.checkRedshiftServerlessWorkgroup(ctx, database, client, meta.RedshiftServerless.WorkgroupName))
+}
+func (c *urlChecker) checkRedshiftServerlessVPCEndpoint(ctx context.Context, database types.Database, client redshiftserverlessiface.RedshiftServerlessAPI, endpointName string) error {
+	endpoint, err := describeRedshiftServerlessVCPEndpoint(ctx, client, endpointName)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return trace.Wrap(requireDatabaseAddressPort(database, endpoint.Address, endpoint.Port))
+}
+func (c *urlChecker) checkRedshiftServerlessWorkgroup(ctx context.Context, database types.Database, client redshiftserverlessiface.RedshiftServerlessAPI, workgroupName string) error {
+	workgroup, err := describeRedshiftServerlessWorkgroup(ctx, client, workgroupName)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if workgroup.Endpoint == nil {
+		return trace.BadParameter("missing endpoint")
+	}
+	return trace.Wrap(requireDatabaseAddressPort(database, workgroup.Endpoint.Address, workgroup.Endpoint.Port))
+}
+
+func (c *urlChecker) checkElastiCache(ctx context.Context, database types.Database) error {
+	meta := database.GetAWS()
+	elastiCacheClient, err := c.clients.GetAWSElastiCacheClient(ctx, meta.Region, cloud.WithAssumeRoleFromAWSMeta(meta))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	cluster, err := describeElastiCacheCluster(ctx, elastiCacheClient, meta.ElastiCache.ReplicationGroupID)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	databases, err := services.NewDatabasesFromElastiCacheReplicationGroup(cluster, nil)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return trace.Wrap(requireContainsDatabaseURLAndType(databases, database, cluster))
+}
+
+func (c *urlChecker) checkMemoryDB(ctx context.Context, database types.Database) error {
+	meta := database.GetAWS()
+	memoryDBClient, err := c.clients.GetAWSMemoryDBClient(ctx, meta.Region, cloud.WithAssumeRoleFromAWSMeta(meta))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	cluster, err := describeMemoryDBCluster(ctx, memoryDBClient, meta.MemoryDB.ClusterName)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return trace.Wrap(requireDatabaseAddressPort(database, cluster.ClusterEndpoint.Address, cluster.ClusterEndpoint.Port))
+}
+
+func (c *urlChecker) checkOpenSearch(ctx context.Context, database types.Database) error {
+	meta := database.GetAWS()
+	client, err := c.clients.GetAWSOpenSearchClient(ctx, meta.Region, cloud.WithAssumeRoleFromAWSMeta(meta))
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	domains, err := client.DescribeDomainsWithContext(ctx, &opensearchservice.DescribeDomainsInput{
+		DomainNames: []*string{aws.String(meta.OpenSearch.DomainName)},
+	})
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if len(domains.DomainStatusList) != 1 {
+		return trace.BadParameter("expect 1 domain but got %v", domains.DomainStatusList)
+	}
+
+	databases, err := services.NewDatabasesFromOpenSearchDomain(domains.DomainStatusList[0], nil)
```

Comment:
Nit: Should we do the same for other databases? Use our `NewDatabaseFromXXX` to compare both databases.

Reply:
we certainly can, and should if we want absolute consistency =D. I prefer just comparing the address directly as we don't need other things from `types.Database` generated by `NewDatabaseXXX`. I only use `NewDatabasessXXX` for the plural cases as otherwise it would be too much logic to compare all URLs generated from one resource.